### PR TITLE
feat: support performance config

### PR DESCRIPTION
**Pull request by inottn**
## Summary

#6226

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/rspack-test-tools/tests/statsOutputCases/performance-different-mode-and-target/webpack.config.js
Comment (LingyuCoder) [inline packages/rspack-test-tools/tests/statsOutputCases/performance-different-mode-and-target/webpack.config.js:1]: Perhaps you can try to reuse webpack test cases directly by removing the `test.filter.js` file.

For example, remove `webpack-test/statsCases/performance-different-mode-and-target/test.filter.js` and run `npm run test:webpack`.

**Comment by chenjiahan** [comment]
Nice 😄

File: packages/rspack-test-tools/tests/statsOutputCases/performance-different-mode-and-target/webpack.config.js
Comment (inottn) [reply packages/rspack-test-tools/tests/statsOutputCases/performance-different-mode-and-target/webpack.config.js:1]: get~

File: webpack-test/statsCases/performance-no-hints/c.js [hunk 1]
```diff
@@ -1 +1,3 @@
-require.ensure(["./d", "./e"], function(require) {});
+// require.ensure(["./d", "./e"], function(require) {});
+import("./d")
```
Comment (LingyuCoder) [inline webpack-test/statsCases/performance-no-hints/c.js:2]: Sorry, my mistake. I didn't notice that these cases used `require.ensure`. It is not recommended to modify `webpack-test`, these cases are used to test rspack's compatibility with webpack. So, it is more appropriate to copy these cases which need to be modified to `rspack-test-tools/tests`.

File: webpack-test/statsCases/performance-different-mode-and-target/webpack.config.js [hunk 2]
```diff
@@ -14,6 +17,9 @@ module.exports = [
 		target: "webworker",
 		output: {
 			filename: "warning.pro-webworker.js"
+		},
+		stats: {
+			hash: false,
```
Comment (LingyuCoder) [inline webpack-test/statsCases/performance-different-mode-and-target/webpack.config.js:22]: If hash is not set to false, what problems will occur?

Comment (inottn) [reply webpack-test/statsCases/performance-different-mode-and-target/webpack.config.js]: I don't know why either. When running `npm run test:webpack`, the generated hashes are different locally and on the CI. https://github.com/web-infra-dev/rspack/actions/runs/8980383687/job/24664828197

Comment (LingyuCoder) [reply webpack-test/statsCases/performance-different-mode-and-target/webpack.config.js]: Caused by hash generating changed on main branch.

**Comment by LingyuCoder** [comment]
Thanks for your contribution

**Review by LingyuCoder — APPROVED** [review]

**Pull request merged**